scripts/golang-version.sh

```diff
@@ -4,12 +4,13 @@ set -x
 
 cd $(dirname $0)
 
+BUILD_BASE_RELEASE_URL=https://api.github.com/repos/rancher/image-build-base/releases
+BUILD_BASE_REQ_HEADERS="Accept: application/vnd.github+json"
 which yq > /dev/null || go install github.com/mikefarah/yq/v4@v4.23.1
 
 K8S_VERSION=$(./semver-parse.sh $1 all)
 DEPENDENCIES_URL="https://raw.githubusercontent.com/kubernetes/kubernetes/${K8S_VERSION}/build/dependencies.yaml"
-GOBORING_RELEASES_URL="https://raw.githubusercontent.com/golang/go/dev.boringcrypto/misc/boring/RELEASES"
 GOLANG_VERSION=$(curl -sL "${DEPENDENCIES_URL}" | yq e '.dependencies[] | select(.name == "golang: upstream version").version' -)
-GOBORING_VERSION=$(curl -sL  "${GOBORING_RELEASES_URL}" | awk "/${GOLANG_VERSION}b.+ [0-9a-f]+ src / {sub(/^go/, \"v\", \$1); print \$1}")
 
-echo ${GOBORING_VERSION}
+GOBORING_TAG=$(curl -s -H ${BUILD_BASE_REQ_HEADERS} ${BUILD_BASE_RELEASE_URL} | jq '.[] | select(.tag_name| contains("'"${GOLANG_VERSION}"'"))' | jq -r .tag_name | head -1)
```

Comment:
```suggestion
GOBORING_TAG=$(curl -s -H ${BUILD_BASE_REQ_HEADERS} ${BUILD_BASE_RELEASE_URL} | jq '[ .[] | select(.tag_name| contains("'"${GOLANG_VERSION}"'")) | .tag_name ] | sort | last'
```

Reply:
for some reason this syntax gshows an error when building

Reply:
updated the comment to add `-r` which omits the quotes on the output string.